src/crypto/slow-hash.c

```diff
@@ -1070,6 +1069,13 @@ void cn_slow_hash(const void *data, size_t length, char *hash)
         hash_extra_blake, hash_extra_groestl, hash_extra_jh, hash_extra_skein
     };
 
+#ifndef FORCE_USE_HEAP
+    uint8_t long_state[MEMORY];
+#else
+    uint8_t *long_state = NULL;
+    long_state = (uint8_t *)malloc(MEMORY);
+#endif
```

Comment:
I don't know if it's a good idea to add c++ code in .c file. Anyway, i tried and std::unique_ptr is not defined (tried to add include memory but file not found)